### DISCUSSION TITLE
why are $ and $$ support not the same as https://math.stackexchange.com/questions/ask ?

### DISCUSSION THREAD

**Discussion by zhiqiangxu** [community/community#45313] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

[This](https://github.com/zhiqiangxu/notes/blob/master/zkSNARK_from_0_to_1.md) is the markdown file I'm writing, when rendered on github, many equations are not displayed correctly(contents inside $$ are not rendered as math).

But when I paste the same content into the editor of https://math.stackexchange.com/questions/ask, it displays smoothly.

Where does the difference come from? Is there any workaround?

**Comment by jge162** [comment]
@zhiqiangxu The difference in rendering is because Stack Exchange editor uses a different typesetting engine than GitHub. GitHub uses CommonMark, which does not support LaTeX-style equations natively. Stack Exchange, on the other hand, uses the MathJax library to render LaTeX equations.

As a workaround, you can use a plugin such as "GitHub-MathJax" to add MathJax support to your GitHub markdown files.

**Reply by zhiqiangxu** [reply]
I installed [this plugin](https://chrome.google.com/webstore/detail/mathjax-plugin-for-github/ioemnmodlmafdkllaclgeombjnmnbima/related?hl=en), but still no effect. Seems github's partial support for `$` and `$$` is making those plugins useless now.

**Comment by nschloe** [comment]
GitHub's math support is not good. I've collected a number of bugs [here](https://github.com/nschloe/github-math-bugs/) and [blogged about it](https://nschloe.github.io/2022/06/27/math-on-github-follow-up.html). The central problem is that the Markdown and math renderers are competing. For the Markdown renderer, `$$xyz$$` is just regular text, and it falsely applies the Markdown rules to it. You don't have these problems with the triple backtick syntax
````
```math
xyz
```
````
but for inline math there's no cure yet. It also seems like this isn't GitHub's priority anymore, at least judging from the fact that their example from the announcment blog post doesn't even work now. You can try [xhub](https://github.com/nschloe/xhub) (a browser extension of mine) -- it's better than GitHub's native support, though of course not perfect either. Perhaps it'll get you somewhere.
